### DISCUSSION TITLE
Auswertung einfach, für Anfänger

### DISCUSSION THREAD

**Discussion by io2345** [tbnobody/OpenDTU#1525] [General]
Habe hier schon ganz tolle Sachen zum Thema Auswertung und Visualisierung gesehen. Ein Anfänger oder technisch weniger versierter OpenDTU Nutzer ist damit aber erst mal überfordert.
Ich würde eine einfache Möglichkeit suchen, zunächst mal rudimentär eine Auswertung über einen längeren Zeitraum zu machen. Ganz ohne zusätzlich laufende Hardware/Server wird's wohl nicht gehen, aber mehr als ein Raspi sollte auch nicht nötig sein.
Gibt es hierfür eine Anleitung, die für Anfänger möglichst frustfrei ist? Reicht ja erst Mal, wenn die Daten der Wechselrichter irgendwo reinfließen und somit nicht verloren sind (dafür ist meines Verständnises nach die MQTT-Schnittstelle da). 
Grafana (ist die grafische Auswertung und Darstellung, richtig?) kann man ja sicherlich auch bei Bedarf auf einem PC aufrufen und sich die Daten vom Speicherort abholen, nehme ich an. Das muss ja nicht permanent vor sich hin laufen, wenn man nur alle paar Tage checken will, was an Ertrag angefallen ist.
Oder ist HomeAssistent schon so ziemlich die einfachste denkbare Lösung? Das entsprechende Image auf einem Raspi zum Laufen zu bringen, würde ich mir jedenfalls gerade noch zutrauen.

**Comment by NintendoTito** [comment]
Home Assistant ist meiner Meinung nach definitiv die einfachste Möglichkeit. Man muss nur Home Assistant auf nem Pi aufsetzen und in der OpenDTU den Punkt MQTT aktivieren und natürlich die IP des HA angeben. Der HA findet dann per Auto-Discovery sämtliche Werte des Wecselrichters über OpenDTU. Dann kann man den Leistungswert dem Energy-Dashboard zufügen und hat die Grafik.

**Comment by tm-107** [comment]
Mein etwas komplizierter Weg (für Anfänger oder technisch nicht versierte User kaum empfehlenswert):
Ein Script auf meinem RaspberryPi liest alle 60 Sekunden die PV-Werte per API aus. Aus den empfangenen JSON-Daten werden dann die gewünschten Daten extrahiert und in eine CSV-Datei gespeichert.
Diese CSV wird dann wiederum auf grafana.net als Source geladen und grafisch dargestellt.
Hier mal der heutige "Ertrag" einer 7,3kW-Anlage:
![Zwischenablage01](https://github.com/tbnobody/OpenDTU/assets/41205746/a1da62c5-39e8-4d85-aa5f-f550872f128a)
(die Temperatur ist der Durchschnittswert von allen 6 Wechselrichtern)

Wie gesagt, das ist nicht der übliche Weg. Aber es geht halt auch ohne InfluxDB, Home Assistand und Co.

Bald folgt hier aber auch ein zusätzlicher RaspberryPi mit InfluxDB, ioBroker und Grafana. Denn wirklich schnell ist die Auswertung der CSV-Datei über grafana.net nicht.

**Reply by io2345** [reply]
@tm-107 : Das würde mir für einen ersten Versuch schon reichen. Magst du das Script zur Verfügung stellen? Die CSV würde ich zunächst lokal speichern und einmal am Tag auf die NAS verschieben oder kopieren.

**Reply by tm-107** [reply]
Mal ganz schnell ohne Anspruch auf Richtigkeit, da ich heute bereits etwas ausgelastet bin ;-)

Ich habe es mal auf das Wesentliche gekürzt, da bei mir noch einige andere Werte abgefragt werden (Kimaanlage, Brauchwasserwärmepumpe, Wallbox, ...).
Angepasst werden müssen eigentlich nur die Variablen am Anfang.
Lese-Zugriffe auf das Webinterface müssen ohne Kennwort möglich sein.

[pv-ertrag.md](https://github.com/tbnobody/OpenDTU/files/13761542/pv-ertrag.md)

Je nach NAS könnte das Script auch direkt darauf laufen ...

Alle 15 Minuten wird die CSV bei mir auf den Websever kopiert. Dabei werden aber nur die vollständigen Werte der letzten 14 Tage sowie die Kopfzeile kopiert. Von allen Tagen davor wird nur der Tagesendwert (23:59 Uhr) übernommen. Ansonsten wird die CSV-Datei irgendwann so groß, dass Grafana die Auswertung abbricht.
Ob das Ganze funktioniert wenn noch keine 14 Tage erfasst sind müsste man mal kurz testen.

[pv-ertrag_copy.md](https://github.com/tbnobody/OpenDTU/files/13761543/pv-ertrag_copy.md)

Viel Erfolg!
